core/src/commonMain/kotlin/ua/syt0r/kanji/core/CoreModule.kt

```diff
@@ -41,7 +42,20 @@ val coreModule = module {
 
     factory<TimeUtils> { DefaultTimeUtils }
 
-    factory<KanjiStrokeEvaluator> { DefaultKanjiStrokeEvaluator() }
+//    factory<KanjiStrokeEvaluator> { DefaultKanjiStrokeEvaluator() }
+
+    // factory produces a *StrokeEvaluator based on settings
+    factory<KanjiStrokeEvaluator> {
+        val preferencesRepository: UserPreferencesRepository= get()
+        var altStrokeEvaluatorEnabled=false
+        runBlocking {altStrokeEvaluatorEnabled=preferencesRepository.getAltStrokeEvaluatorEnabled()}
```

Comment:
I thought I'd use the factory since it was already there. I did think about making a switch in writing practice config but then "realized" that it might not work because the stroke evaluator would already instantiated? Does that make sense? TBH, I am more used to tight computational loops in C and only doing Kotlin for two weeks. Not sure what to do here. Would appreciate your help on this one.

Reply:
This place is adding `KanjiStrokeEvaluator` to the dependency injection graph of the application. But I want you to remove evaluator from the constructor of `WritingPracticeViewModel` and make it a private lateinit field that you will initialize inside of `onPracticeConfigured` method. Stroke evaluators don't have any dependencies so they can be made kotlin's `object` instead of `class`, basically a singletons, so you'll just assign the necessary one there. Also since dependency injection won't be necessary to provide evaluators this part with factory can be removed

And move the checkbox from the frdoid settings screen to the `ConfiguringState` composable function of writing practice screen, place it in the end of the list